controllers/k8ssandra/stargate.go

```diff
@@ -105,7 +105,18 @@ func (r *K8ssandraClusterReconciler) reconcileStargate(
 }
 
 // TODO move to stargate package
-func (r *K8ssandraClusterReconciler) newStargate(stargateKey types.NamespacedName, kc *api.K8ssandraCluster, stargateTemplate *stargateapi.StargateDatacenterTemplate, actualDc *cassdcapi.CassandraDatacenter) *stargateapi.Stargate {
+func (r *K8ssandraClusterReconciler) newStargate(stargateKey types.NamespacedName, kc *api.K8ssandraCluster, stargateTemplate *stargateapi.StargateDatacenterTemplate, actualDc *cassdcapi.CassandraDatacenter, dcTemplate api.CassandraDatacenterTemplate) *stargateapi.Stargate {
+	cassandraEncryption := stargateapi.CassandraEncryption{}
+	if dcTemplate.CassandraConfig != nil {
+		if dcTemplate.CassandraConfig.CassandraYaml.ClientEncryptionOptions != nil && kc.Spec.Cassandra.ClientEncryptionStores != nil {
```

Comment:
What if `ClientEncryptionOptions` is present but `ClientEncryptionOptions.Enabled = false`? Cf. the function `ClientEncryptionEnabled` – maybe you should use it here.

Reply:
switched to using `ClientEncryptionEnabled`